src/components/KTableData/KTableData.vue

```diff
@@ -98,65 +100,61 @@
     <template
       v-for="slot in getHeaderSlots"
       :key="slot"
-      #[slot]="{ column }"
+      #[slot]="slotProps"
     >
       <slot
-        :column="column"
         :name="slot"
+        v-bind="slotProps"
       />
     </template>
 
     <template
       v-for="slot in getHeaderTooltipSlots"
       :key="slot"
-      #[slot]="{ column }"
+      #[slot]="slotProps"
     >
       <slot
-        :column="column"
         :name="slot"
+        v-bind="slotProps"
       />
     </template>
 
     <template
       v-for="slot in getCellSlots"
       :key="slot"
-      #[slot]="{ row, rowKey, rowValue }"
+      #[slot]="slotProps"
     >
       <slot
         :name="slot"
-        :row="row"
-        :row-key="rowKey"
-        :row-value="rowValue"
+        v-bind="slotProps"
       />
     </template>
 
     <template
-      v-if="$slots['action-items']"
-      #action-items="{ row }"
+      v-if="slots['action-items']"
+      #action-items="slotProps"
     >
       <slot
         name="action-items"
-        :row="row"
+        v-bind="slotProps"
       />
     </template>
 
     <template
-      v-if="$slots['row-expanded']"
-      #row-expanded="{ columnWidths, nestedHeaders, row }"
+      v-if="slots['row-expanded']"
+      #row-expanded="slotProps"
     >
       <slot
-        :column-widths="columnWidths"
         name="row-expanded"
-        :nested-headers="nestedHeaders"
-        :row="row"
+        v-bind="slotProps"
       />
     </template>
   </KTableView>
 </template>
 
-<script setup lang="ts">
+<script setup lang="ts" generic="const H extends TableDataHeader = TableDataHeader, D extends readonly Record<string, any>[] = readonly Record<string, any>[], O extends string | number = string | number">
```

Comment:
nit: could we name the generics a bit more specifically so it's easier to read down below?